### ISSUE TITLE
Can't delete objects from s3 bucket using golang SDK

### ISSUE THREAD

**Issue by enzo-beltrami-ifood** [localstack/localstack#2850]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description
When I try to delete an item from aws localstack s3 bucket I get this error:
```
SerializationError: failed to unmarshal error message
        status code: 500, request id: E7B17F98419A4F32, host id: MzRISOwyjmnupE7B17F98419A4F327/JypPGXLh0OVFGcJaaO3KW/hRAqKOpIEEp
caused by: UnmarshalError: failed to unmarshal error message
00000000  3c 21 44 4f 43 54 59 50  45 20 48 54 4d 4c 20 50  |<!DOCTYPE HTML P|
00000010  55 42 4c 49 43 20 22 2d  2f 2f 57 33 43 2f 2f 44  |UBLIC "-//W3C//D|
00000020  54 44 20 48 54 4d 4c 20  33 2e 32 20 46 69 6e 61  |TD HTML 3.2 Fina|
00000030  6c 2f 2f 45 4e 22 3e 3c  74 69 74 6c 65 3e 35 30  |l//EN"><title>50|
00000040  30 20 49 6e 74 65 72 6e  61 6c 20 53 65 72 76 65  |0 Internal Serve|
00000050  72 20 45 72 72 6f 72 3c  2f 74 69 74 6c 65 3e 3c  |r Error</title><|
00000060  68 31 3e 49 6e 74 65 72  6e 61 6c 20 53 65 72 76  |h1>Internal Serv|
00000070  65 72 20 45 72 72 6f 72  3c 2f 68 31 3e 3c 70 3e  |er Error</h1><p>|
00000080  54 68 65 20 73 65 72 76  65 72 20 65 6e 63 6f 75  |The server encou|
00000090  6e 74 65 72 65 64 20 61  6e 20 69 6e 74 65 72 6e  |ntered an intern|
000000a0  61 6c 20 65 72 72 6f 72  20 61 6e 64 20 77 61 73  |al error and was|
000000b0  20 75 6e 61 62 6c 65 20  74 6f 20 63 6f 6d 70 6c  | unable to compl|
000000c0  65 74 65 20 79 6f 75 72  20 72 65 71 75 65 73 74  |ete your request|
000000d0  2e 20 45 69 74 68 65 72  20 74 68 65 20 73 65 72  |. Either the ser|
000000e0  76 65 72 20 69 73 20 6f  76 65 72 6c 6f 61 64 65  |ver is overloade|
000000f0  64 20 6f 72 20 74 68 65  72 65 20 69 73 20 61 6e  |d or there is an|
00000100  20 65 72 72 6f 72 20 69  6e 20 74 68 65 20 61 70  | error in the ap|
00000110  70 6c 69 63 61 74 69 6f  6e 2e 3c 2f 70 3e 0a     |plication.</p>.|
```
...

## Expected behavior
deleteObject operation should delete the object
...

## Actual behavior
deleteObject throws an error 
...

# Steps to reproduce

## Command used to start LocalStack
```
localstack start
awslocal s3api create-bucket --bucket test
```
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```go
package main

import (
	"fmt"
	"strings"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
	"github.com/aws/aws-sdk-go/service/s3/s3manager"
)

func main() {
	sessionOptions := session.Options{
		Config: aws.Config{
			Region:           aws.String("us-east-1"),
			Endpoint:         aws.String("http://localhost:4566"),
			S3ForcePathStyle: aws.Bool(true),
		}}

	sess := session.Must(session.NewSessionWithOptions(sessionOptions))
	uploadSvc := s3manager.NewUploader(sess)
	s3Svc := s3.New(sess)

	bucket := aws.String("test")
	key := aws.String("test.json")
	data := "{\"test\": \"data\"}"

	// Works fine
	_, err := uploadSvc.Upload(&s3manager.UploadInput{
		Bucket: bucket,
		Key:    key,
		Body:   strings.NewReader(data),
	})

	if err != nil {
		fmt.Println(err)
		return
	}

	// works on aws but doesnt work on localstack
	_, err = s3Svc.DeleteObjects(&s3.DeleteObjectsInput{
		Bucket: bucket,
		Delete: &s3.Delete{
			Objects: []*s3.ObjectIdentifier{{Key: key}},
		},
	})

	if err != nil {
		fmt.Println(err)
		return
	}
}

```
...

**Comment by enzo-beltrami-ifood** [comment]
I wrote similar code using nodejs and it works just fine, seems to be an issue with golang SDK

**Comment by whummer** [comment]
Thanks for reporting @enzo-beltrami-ifood @yaitskov . This should be fixed in #2868 - can you please pull the `latest` image and give it another try? Please report here if the problem persists. Thanks
